spanner/api/Spanner/Program.cs

```diff
@@ -1757,35 +1757,44 @@ await connection.BeginTransactionAsync())
-                    LastName = "Richards"},
-                new Singer {SingerId = secondSingerId, FirstName = "Catalina",
-                    LastName = "Smith"},
-                new Singer {SingerId = 3, FirstName = "Alice",
-                    LastName = "Trentor"},
-                new Singer {SingerId = 4, FirstName = "Lea",
-                    LastName = "Martin"},
-                new Singer {SingerId = 5, FirstName = "David",
-                    LastName = "Lomond"},
+            List<Album> albums = new List<Album> {
+                new Album { SingerId = secondSingerId, AlbumId = 1, AlbumTitle = "Green" },
+                new Album { SingerId = secondSingerId, AlbumId = 3, AlbumTitle = "Terrified" },
             };
             // Create connection to Cloud Spanner.
             using (var connection = new SpannerConnection(connectionString))
             {
                 await connection.OpenAsync();
 
-                // Delete rows from the Singers table.
-                var cmd = connection.CreateDeleteCommand("Singers",
+                // Delete individual rows from the Albums table.
+                var cmd = connection.CreateDeleteCommand(
+                    "Albums",
                     new SpannerParameterCollection {
-                        {"SingerId", SpannerDbType.Int64}
-                    });
-                await Task.WhenAll(singers.Select(singer =>
+                        {"SingerId", SpannerDbType.Int64},
+                        {"AlbumId", SpannerDbType.Int64}
+                    }
+                );
+                await Task.WhenAll(albums.Select(album =>
                 {
-                    cmd.Parameters["SingerId"].Value = singer.SingerId;
+                    cmd.Parameters["SingerId"].Value = album.SingerId;
+                    cmd.Parameters["AlbumId"].Value = album.AlbumId;
                     return cmd.ExecuteNonQueryAsync();
                 }));
 
-                Console.WriteLine("Deleted data.");
+                Console.WriteLine("Deleted individual rows in Albums.");
+
+                // Delete a range of rows from the Singers table where the column key is >=3 and <5.
+                cmd = connection.CreateDmlCommand(
+                   "DELETE FROM Singers WHERE SingerId >= 3 AND SingerId < 5");
+                int rowCount = await cmd.ExecuteNonQueryAsync();
+                Console.WriteLine($"{rowCount} row(s) deleted from Singers.");
+
+                // Delete remaining Singers rows, which will also delete the remaining Albums rows
+                // since it was defined with ON DELETE CASCADE.
+                cmd = connection.CreateDmlCommand(
+                   "DELETE FROM Singers WHERE true");
+                rowCount = await cmd.ExecuteNonQueryAsync();
+                Console.WriteLine($"{rowCount} row(s) deleted from Singers.");
             }
         }
         // [END spanner_delete_data]
```

Comment:
This sample now does a lot of things. Would it be possible to split it in several methods, even if they are all enclosed by the same region tag. I'm thinking on testability, specially when we move away from command line runners, we would like to test against something other than the standard output, etc.

Reply:
I separated it out into their own functions. I would like to keep them within a single region tag when we move it out to separate files since that would make it easier to pull the samples into the docs. Hope that's ok.

Reply:
Yes, that's totally fine, there's nothing wrong with having multiple methods, or even type definitions, in the same region tag. A region tag though should be fully defined in a single file, else it's really hard for maintainers to a) follow what a sample (the code that's actually shown) does and b) test what a sampel does.
It's also fine to have several tags show exactly the same code. The case might be that language 1 does things different for sample a and sample b, but for language 2, samples a and b are exactly the same. You would have two region tags, in language 1 you have to separate samples for each of those tags, in language 2 you have the same sample enclosed by both tags.